From 8c13923a8e17a02be0989649b2edc20124816729 Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Tue, 15 Jun 2021 16:08:22 -0400
Subject: [PATCH] seccomp: undef fstatat64 to avoid build failure on musl

sys/stat.h in musl does this:

  #define fstatat64 fstatat

Counteract this with an #undef.

Bug: https://bugs.gentoo.org/789336
Bug: https://bugs.astron.com/view.php?id=473
Signed-off-by: Mike Gilbert <floppym@gentoo.org>
Signed-off-by: Waldemar Brodkorb <wbx@openadk.org>
Upstream: N/A under discussion
---
 src/seccomp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/seccomp.c b/src/seccomp.c
index 5a39ee45..d2a1139a 100644
--- a/src/seccomp.c
+++ b/src/seccomp.c
@@ -205,6 +205,7 @@ enable_sandbox_full(void)
 #endif
  	ALLOW_RULE(fstat64);
 #ifdef __NR_fstatat64
+#undef fstatat64
 	ALLOW_RULE(fstatat64);
 #endif
 	ALLOW_RULE(futex);
